ecl/hthor/hthor.cpp

```diff
@@ -2494,14 +2494,62 @@ const void *CHThorGroupDedupKeepRightActivity::nextRow()
 
 //=====================================================================================================
 
+CHThorGroupDedupKeepBestActivity::CHThorGroupDedupKeepBestActivity(IAgentContext &_agent, unsigned _activityId, unsigned _subgraphId, IHThorDedupArg &_arg, ThorActivityKind _kind, bool _keepLeft)
+    : CHThorGroupDedupActivity(_agent, _activityId, _subgraphId, _arg, _kind), keepLeft (_keepLeft)
+{
+    compareBest = helper.queryCompareBest();
+}
+
+void CHThorGroupDedupKeepBestActivity::ready()
+{
+    CHThorGroupDedupActivity::ready();
+    assertex(numToKeep==1);
```

Comment:
See prev. question re. BEST + KEEP(n) ..
semantically I think would make sense to support both? i.e. essentially means BEST N